### PR TITLE
Surgery Overhaul

### PR DESCRIPTION
**Pull request by Geevies**
Polishes up Surgery for the upcoming Medical overhaul.

* Surgery sites now have icon overlays, you can now physically see incisions and retracted ribcages.
* Roller beds have been buffed to be 80% safe to operate on, up from 75%
* Using surgery tools on unsafe surfaces (rollerbeds, tables) no longer ATTACKS the patient.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by KingOfThePing** [comment]
nut

File: code/_helpers/unsorted.dm [hunk 1]
```diff
@@ -898,11 +898,24 @@ proc/is_hot(obj/item/W as obj)
...
 //check if mob is lying down on something we can operate him on.
-/proc/can_operate(mob/living/carbon/M)
-	return (M.lying && \
-	locate(/obj/machinery/optable, M.loc) || \
-	(locate(/obj/structure/bed/roller, M.loc) && prob(75)) || \
-	(locate(/obj/structure/table/, M.loc) && prob(66)))
+/proc/can_operate(mob/living/carbon/M) //If it's 2, commence surgery, if it's 1, fail surgery, if it's 0, attack
+	if(M.lying)
+		if(locate(/obj/machinery/optable, M.loc))
+			world << "2"
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm]: hmm

File: code/modules/surgery/organs_internal.dm [hunk 2]
```diff
@@ -24,85 +23,83 @@
...
-			dam_amt = 5
-			target.adjustToxLoss(10)
-			affected.createwound(CUT, 5)
+	for(var/obj/item/organ/I in affected.internal_organs)
+		if(I && I.damage > 0)
+			if(I.robotic < 2)
+				user.visible_message("<span class='notice'>[user] treats damage to [target]'s [I.name] with [tool_name].</span>", \
+					"<span class='notice'>You treat damage to [target]'s [I.name] with [tool_name].</span>" )
+				I.damage = 0
+				I.status &= ~ORGAN_BROKEN
```
Comment (NonQueueingMatt) [inline code/modules/surgery/organs_internal.dm]:
```suggestion
				
```
this doesn't fix it

File: code/_helpers/unsorted.dm (hunk 1 above)
Comment (Geevies) [inline code/_helpers/unsorted.dm]:
```suggestion
```

File: code/_helpers/unsorted.dm [hunk 3]
```diff
@@ -898,11 +898,23 @@ proc/is_hot(obj/item/W as obj)
...
 //check if mob is lying down on something we can operate him on.
-/proc/can_operate(mob/living/carbon/M)
-	return (M.lying && \
-	locate(/obj/machinery/optable, M.loc) || \
-	(locate(/obj/structure/bed/roller, M.loc) && prob(75)) || \
-	(locate(/obj/structure/table/, M.loc) && prob(66)))
+/proc/can_operate(mob/living/carbon/M) //If it's 2, commence surgery, if it's 1, fail surgery, if it's 0, attack
+	if(M.lying)
+		if(locate(/obj/machinery/optable, M.loc))
+			return 2
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm]: these should be defines + you can condense all this shit into one return

File: code/_onclick/item_attack.dm [hunk 4]
```diff
@@ -37,10 +37,19 @@ avoid code duplication. This includes items that may sometimes act as a standard
...
 /mob/living/attackby(obj/item/I, mob/user)
 	if(!ismob(user))
-		return 0
-	if(can_operate(src) && do_surgery(src,user,I)) //Surgery
-		return 1
-	return I.attack(src, user, user.zone_sel.selecting)
+		return FALSE
+
+	var/operating = can_operate(src)
+	if(operating == 2) //Safe surgery
```
Comment (NonQueueingMatt) [inline code/_onclick/item_attack.dm]: defines

**Pull request merged**